src/parquet/file/reader-internal.cc

```diff
@@ -17,17 +17,138 @@
 
 #include "parquet/file/reader-internal.h"
 
-#include <memory>
+#include <string.h>
+#include <algorithm>
+#include <exception>
+#include <ostream>
 #include <vector>
 
-#include "parquet/column/serialized-page.h"
+#include "parquet/column/page.h"
+#include "parquet/compression/codec.h"
+#include "parquet/exception.h"
 #include "parquet/schema/converter.h"
+#include "parquet/schema/descriptor.h"
+#include "parquet/schema/types.h"
 #include "parquet/thrift/util.h"
+#include "parquet/types.h"
 #include "parquet/util/input.h"
 
 namespace parquet_cpp {
 
 // ----------------------------------------------------------------------
+// SerializedPageReader deserializes Thrift metadata and pages that have been
+// assembled in a serialized stream for storing in a Parquet files
+
+SerializedPageReader::SerializedPageReader(std::unique_ptr<InputStream> stream,
+    Compression::type codec) :
```

Comment:
Should we name this `ThriftSerializedPageReader` since it is specific to thrift? If we want to move on the lines of `parquet-mr`, I believe there should be further de-coupling from thrift and this Reader. We probably should have another module like `parquet-cpp/parquet-thrift`.